### PR TITLE
Storm hamilton code review

### PR DESCRIPTION
**Pull request by hamilsu**
So there are a few things that I want to suggest. 

First of all, there was almost no error checking for setting any of the wages and hours worked, specifically if a user inputted a negative value. I added some setters that should override the lombok ones and default the values to zero if they are negative.

Secondly, there was nothing in place to insure that any monetary values were in a currency format (as in they're set to the second decimal place.) I have added a round function to the payroll class and rewrote the calculateAmountPaid class to use it. 

Additionally, I added in autowiring for payroll service

Finally, I noticed a lack of javadoc. I went around and added a few instances of it, especially for any methods I changed. I recommend adding the rest of the javadoc when you get the time.

Overall the program compiled and looked fine for the one page that was visible. Obviously there is some more development that needs to happen for the app to be functional. It was pretty straight forward to understand, nothing too complicated.

### CODE REVIEW

**Review by vumd-UC — APPROVED** [review]
I think all of these changes are fantastic, and should be merged into our project.
@jros07 @JayBit33

File: src/main/java/com/timekeeper/app/controllers/TimeKeeperController.java [hunk 1]
```diff
@@ -24,7 +24,8 @@
 @Controller
 public class TimeKeeperController {
 
-    PayrollService payrollService = new PayrollService();
+    @Autowired
+    IPayrollService payrollService;
```
Comment (discospiff) [inline src/main/java/com/timekeeper/app/controllers/TimeKeeperController.java:28]: Definitely... good idea.  That's the whole point of Spring Boot.  :)

File: src/main/java/com/timekeeper/app/dao/DataStorage.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.timekeeper.app.dao;
 
 /**
- * Dao is our persistance layer
+ * Dao is our persistence layer
```
Comment (discospiff) [inline src/main/java/com/timekeeper/app/dao/DataStorage.java:4]: Good catch.

File: src/main/java/com/timekeeper/app/dto/Employee.java [hunk 3]
```diff
@@ -11,6 +11,19 @@
...
 
+
+    /**
+     * Sets the employee's wage after making sure the value isn't a negative number
+     * @param wage A double that reflects the employee's wage.
+     */
+    public void setWage(double wage){
+        if (wage<0.0){
+        wage = 0.0;
+        }
```
Comment (discospiff) [inline src/main/java/com/timekeeper/app/dto/Employee.java:22]: This is a good time to throw an exception.  Tell the calling method that something is wrong.

Also, indent.  Control Alt I will do it for you automatically in IntelliJ IDEA.

File: src/main/java/com/timekeeper/app/dto/Payroll.java [hunk 4]
```diff
@@ -12,14 +15,46 @@
...
+     * @param places the number of decimal places to be rounded to
+     * @return an appropriately rounded double
+     */
+    public static double round(double value, int places) {
+        if (places < 0) throw new IllegalArgumentException();
+
+        BigDecimal bd = BigDecimal.valueOf(value);
+        bd = bd.setScale(places, RoundingMode.HALF_UP);
+        return bd.doubleValue();
+    }
```
Comment (discospiff) [inline src/main/java/com/timekeeper/app/dto/Payroll.java:44]: I'd make one more (very important) change.  As long as you're using a BigDecimal, use it as the return type as well.  doubles are subject to Floating Point Arithmetic, which can lead to inaccuracies.  BigDecimal is not.  So, as long as you're using BigDecimal, use it universally.

**Comment by discospiff** [comment]
Several good suggestions that reduce technical debt.
I recommend merging these changes.

**Comment by JayBit33** [comment]
Because we waited so long to merge there are hundreds of new lines of code and now merge conflicts I implemented suggestions by hand. Great review. Thank You!

**Pull request closed without merging**